packages/parse5/lib/parser/index.test.ts

```diff
@@ -98,4 +100,29 @@ describe('parser', () => {
         expect(doctype).toHaveProperty('publicId', '');
         expect(doctype).toHaveProperty('systemId', '');
     });
+
+    describe('Tree adapters', () => {
+        it('should support onItemPush and onItemPop', () => {
+            const onItemPush = jest.fn();
+            const onItemPop = jest.fn();
+            const document = parse5.parse('<p><p>', {
+                treeAdapter: {
+                    ...treeAdapters.default,
+                    onItemPush,
+                    onItemPop,
+                },
+            });
+
+            const htmlElement = document.childNodes[0];
+            assert.ok(isElementNode(htmlElement));
+            const bodyElement = htmlElement.childNodes[1];
+            assert.ok(isElementNode(bodyElement));
+            expect(onItemPush).toHaveBeenCalledTimes(5);
+            expect(onItemPush).toHaveBeenNthCalledWith(1, htmlElement);
+            expect(onItemPush).toHaveBeenNthCalledWith(3, bodyElement);
+            expect(onItemPush).toHaveBeenLastCalledWith(bodyElement.childNodes[1]);
+            expect(onItemPop).toHaveBeenCalledTimes(2);
+            expect(onItemPop).toHaveBeenLastCalledWith(bodyElement.childNodes[0], bodyElement);
```

Comment:
Looks a bit like magic numbers. Perhaps good to comment why 5, 1, 3, 2?

Reply:
Added some comments.